frontend/js/src/artist/ArtistPage.tsx

```diff
@@ -84,11 +117,52 @@ export default function ArtistPage(): JSX.Element {
     WikipediaExtract
   >();
 
-  const [albumsByThisArtist, alsoAppearsOn] = partition(
+  const [sort, setSort] = React.useState<"release_date" | "total_listen_count">(
+    "release_date"
+  );
+
+  const rgGroups = groupBy(
     releaseGroups,
-    (rg) => rg.artists[0].artist_mbid === artist?.artist_mbid
+    (rg) =>
+      rg.type + (rg.secondary_types?.[0] ? ` + ${rg.secondary_types?.[0]}` : "")
```

Comment:
Annoyingly, primary type can be undefined while secondary type isn't (valid in MB) and results in this:
![image](https://github.com/user-attachments/assets/8104b45e-9616-4b34-9fe1-6197c262efb8)

Because there is a secondary type, the bit of code where you check `rgGroups.undefined` on line 143 does not apply to this `undefined + SecondaryType`.

So maybe default to primary type to Other to resolve it?
Not sure the check on line 143 will still be useful after that.
```suggestion
      (rg.type ?? "Other") + (rg.secondary_types?.[0] ? ` + ${rg.secondary_types?.[0]}` : "")
```